server/src/main/java/io/crate/legacy/LegacySettings.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.legacy;
+
+import org.elasticsearch.common.settings.Setting;
+
+public class LegacySettings {
+
+    private static final String TABLE_FUNCTION_COLUMN_NAMING = "legacy.table_function_column_naming";
+
+    public static final Setting<Boolean> LEGACY_TABLE_FUNCTION_COLUMN_NAMING = Setting.boolSetting(
+        TABLE_FUNCTION_COLUMN_NAMING, false, Setting.Property.NodeScope);
```

Comment:
Thanks for bringing it up, I do not have a strong opinion for this.
Looks like node settings are always static and cluster settings can be runtime. @mfussenegger suggested this to be node-level setting, could you clarify? ccing @seut as Jordi is away

Reply:
The reason for having it non-dynamic and such only changeable by a node restart was I guess just for simplicity. Supporting that it can be change at runtime would require much more effort as the relevant setting could not be simply passed to the function registry but a setting consumer etc would have to be implemented.
I see no strong reason to support this at runtime, I guess in most cases this is not required.

Btw. even that this is flagged with `NodeScope` this must be documented as a cluster setting. It must be the same for all nodes otherwise it will result in non-deterministic and weird behaviour (function would have different return type per node).

Reply:
I think having different settings per node would not harm, but if a statement works or not (setting kicks in and returned name changes) would then depend on the node the statement is sent to (handler node).

Reply:
> I think having different settings per node would not harm, but if a statement works or not (setting kicks in and returned name changes) would then depend on the node the statement is sent to (handler node).

Added an extra note,
```
  .. NOTE::

    Beware that if not all nodes in the cluster are consistently set or unset,
    the behaviour will depend on the node handling the query.
```

Reply:
That was a concern as well, to allow the possibility to have nodes behaving differently, depending on their individual setting, but since this a bwc flag, I don't insist at all to introduce complexity with a dynamic cluster wide setting.